keras/backend/jax/nn.py

```diff
@@ -622,3 +622,73 @@ def _iterate(prev, x):
     )
 
     return -last_alpha_mask[jnp.arange(batch_size), target_length]
+
+
+def ctc_greedy_decode(
+    inputs,
+    sequence_length,
+    merge_repeated=True,
+    mask_index=None,
+):
+    max_class_indices = jnp.argmax(inputs, axis=-1)
+
+    max_class_scores = jnp.max(inputs, axis=-1)
+    max_class_scores = jnp.where(
+        max_class_indices == mask_index, 0.0, max_class_scores
+    )
+
+    if mask_index is None:
+        mask_index = -1
+
+    if merge_repeated:
+        repeat = max_class_indices[:, 1:] == max_class_indices[:, :-1]
+        repeat = repeat & (max_class_indices[:, 1:] != mask_index)
+        repeat = jnp.pad(repeat, ((0, 0), (1, 0)))
+
+        max_class_indices = jnp.where(repeat, mask_index, max_class_indices)
+
+        seqlen_mask = jnp.arange(inputs.shape[1])[None, :] > sequence_length
+        max_class_indices = jnp.where(
+            seqlen_mask, mask_index, max_class_indices
+        )
+
+        def _iterate(prev, x):
+            prev_score, prev_repeat = prev
+            score, repeat = x
+            new_score = score + jnp.where(prev_repeat, prev_score, 0.0)
+            return (new_score, repeat), new_score
+
+        _, max_class_scores = lax.scan(
+            _iterate,
+            (
+                jnp.zeros((inputs.shape[0],)),
+                jnp.zeros((inputs.shape[0],), dtype=bool),
+            ),
+            (max_class_scores.transpose(), repeat.transpose()),
+            reverse=True,
+        )
+        max_class_scores = max_class_scores.transpose()
+
+    final_indices = jnp.where(max_class_indices != mask_index)
+    max_class_indices = max_class_indices[final_indices]
+    max_class_scores = max_class_scores[final_indices]
+
+    return max_class_indices, -max_class_scores
+
+
+def ctc_decode(
+    inputs,
+    sequence_length,
+    strategy,
+    beam_width=100,
+    top_paths=1,
+    merge_repeated=True,
+    mask_index=None,
+):
+    if strategy == "greedy":
```

Comment:
This would need the fallback case `else:`

Reply:
Yes, added, thanks.